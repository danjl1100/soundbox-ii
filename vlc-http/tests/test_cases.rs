// Copyright (C) 2021-2024  Daniel Lambert. Licensed under GPL-3.0-or-later, see /COPYING file for details

//! Test case runner for `vlc_http`

// yes panic, it's tests!
#![allow(clippy::panic)]

use test_log::test;
use tracing::{trace_span, warn_span};

mod common;

#[test]
fn test_cases() {
    insta::glob!("inputs/*.txt", test_case);
}

fn test_case(input: &std::path::Path) {
    let name = input
        .file_name()
        .expect("input file has a name")
        .to_str()
        .expect("test input filenames are UTF-8");
    let span = warn_span!("test_case", %name);
    let _guard = span.enter();

    let span = trace_span!("test_case", input_path=%input.display());
    let _guard = span.enter();

    let input = std::fs::read_to_string(input).expect("test input file exists");
    let output = common::run_input(&input);
    insta::assert_ron_snapshot!(output);
}
