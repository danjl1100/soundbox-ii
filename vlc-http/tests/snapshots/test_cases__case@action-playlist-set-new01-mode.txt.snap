---
source: vlc-http/tests/test_cases.rs
expression: output
input_file: vlc-http/tests/inputs/action-playlist-set-new01-mode.txt
---
[
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json",
  ), Model()),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_random",
  ), Model(
    is_random: true,
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_repeat",
  ), Model(
    is_repeat_one: true,
    is_random: true,
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json",
  ), Model(
    is_repeat_one: true,
    is_random: true,
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_random",
  ), Model(
    is_repeat_one: true,
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_repeat",
  ), Model()),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json",
  ), Model()),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json?command=in_enqueue&input=file%3A%2F%2Fa%2F",
  ), Model(
    items: {
      0: "file://a/",
    },
  )),
]
